eslint.config.mjs

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+/* eslint-disable import/no-unresolved */
+/* eslint-disable no-underscore-dangle */
+import { fileURLToPath } from 'node:url'
+import { FlatCompat } from '@eslint/eslintrc'
+import eslint from '@eslint/js'
+import path from 'node:path'
+import tseslint from 'typescript-eslint'
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: eslint.configs.recommended,
+  allConfig: eslint.configs.all,
+  ...tseslint.configs.recommended,
+})
+
+export default tseslint.config(
+  {
+    ignores: [
+      '**/node_modules/',
+      '**/vendor/',
+      '**/build/',
+      '**/coverage/',
+      '**/dist/',
+      '**/tmp/',
+      'proprietary/react-icons',
+      'packages/react/unstyled',
```

Comment:
These files are not in this repository – the config was probably copied from the design-system repo.